app/controllers/search_controller.rb

```diff
@@ -1,4 +1,16 @@
 class SearchController < ApplicationController
   def index
+    @facade = OomprBeFacade.new
+    @listings = @facade.search_listings_by_city(search_params[:city])
+    if @listings.empty?
+      flash[:notice] = "No listings found. Please try again." 
+      redirect_to root_path
+    end
```

Comment:
great job including this flash notice - meant to ask about it in a meeting. I didn't account for this error handling on the backend, because it felt like more of a frontend sad path